src/OrbitGl/ScopedStatus.h

```diff
@@ -37,11 +37,14 @@ class ScopedStatus final {
  public:
   ScopedStatus() = default;
   explicit ScopedStatus(std::weak_ptr<orbit_base::Executor> executor,
-                        StatusListener* status_listener, const std::string& status_message) {
+                        std::weak_ptr<StatusListener> maybe_status_listener,
+                        const std::string& status_message) {
     data_ = std::make_unique<Data>();
     data_->executor = std::move(executor);
-    data_->status_listener = status_listener;
+    data_->status_listener = std::move(maybe_status_listener);
     data_->main_thread_id = std::this_thread::get_id();
+
+    std::shared_ptr<StatusListener> status_listener = data_->status_listener.lock();
     data_->status_id = status_listener->AddStatus(status_message);
```

Comment:
Couldn't `status_listener` be a nullptr at this point if you already pass in an expired weak_ptr? IMO you should at least check for this in the next line, as the parameter name suggests this can handle expired pointers.

Reply:
Yes. True. Fixed.